tests/approx_matching_ut.cpp

```diff
@@ -201,50 +201,91 @@ SIMPLE_UNIT_TEST_SUITE(ApproxMatchingTest) {
 		}
 	}
 
+	enum MutateOperation {
+		Substitute = 0,
```

Comment:
If values of a enum are not important they can be assigned automatically.
Moreover, it is easy to define bounds for `for` loops over enum values automatically.
Eg.:
```
enum MutateOperation {
        Begin,
        Substitute = Begin,
        Delete,
        Insert,
        End,
};
```
This allows one to write:
```
for (int operation = MutateOperation::Begin; operation < MutateOperation::End; ++operation) {
```
And if some operations are added to the enum just above the `End` later, `for` loops does not need to be modified.
So it is impossible to forget to modify the loops.